packages/manager/apps/pci-instances/src/data/hooks/catalog/selectors/models.selector.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  TCpuDto,
+  TMemoryDto,
+  TGpuDto,
+  TDiskDto,
+  TSpecificationsDto,
+  TPricingDto,
+  TModelDto,
+  TCatalogDto,
+} from '@/types/catalog/api.types';
+import {
+  TCpu,
+  TMemory,
+  TGpu,
+  TStorage,
+  TModelSpecification,
+  TModelPricing,
+  TModel,
+  TModelEntity,
+} from '@/types/catalog/entity.types';
+import { DeepReadonly } from '@/types/utils.type';
+
+export type TModelSelector = (rawData: TCatalogDto) => TModelEntity;
+
+const getSizeUnit = (size: number) => (size < 1000 ? 'Go' : 'To');
+
+const mapModelCpu = (cpu: TCpuDto): TCpu => {
+  const { type, ...rest } = cpu;
+  return rest;
+};
+
+const mapModelMemory = (memory: TMemoryDto): TMemory => ({
+  size: memory.size,
+  unit: getSizeUnit(memory.size),
+});
+
+const mapModelGpu = ({ model, number }: TGpuDto): TGpu => ({
+  model,
+  number,
+});
+
+const mapModelStorages = (
+  storages: DeepReadonly<TDiskDto[]>,
+): DeepReadonly<TStorage[]> =>
+  storages.map((storage) => ({
+    ...storage,
+    sizeUnit: getSizeUnit(storage.capacity),
+  }));
+
+const mapModelSpecifications = (
+  specifications: DeepReadonly<TSpecificationsDto>,
+): DeepReadonly<{ specifications: TModelSpecification }> => ({
+  specifications: {
+    memory: mapModelMemory(specifications.memory),
+    cpu: mapModelCpu(specifications.cpu),
+    bandwidth: specifications.bandwidth.level,
+    storage: mapModelStorages(specifications.storage.disks),
+    gpu: mapModelGpu(specifications.gpu),
+  },
+});
+
+const mapModelPricings = (
+  pricings: DeepReadonly<TPricingDto[]>,
+): DeepReadonly<{ pricings: TModelPricing[] }> => ({
+  pricings: pricings
+    .reduce(
+      (acc: DeepReadonly<TPricingDto[]>, cur: DeepReadonly<TPricingDto>) => {
+        if (!acc.length) return [cur];
+
+        const foundPricingByInterval = acc.find(
+          (elt) => elt.interval === cur.interval,
+        );
+
+        if (!foundPricingByInterval) return [...acc, cur];
+        if (foundPricingByInterval.price < cur.price) return acc;
+        return [
+          ...acc.filter((elt) => elt.price !== foundPricingByInterval.price),
+          cur,
+        ];
+      },
+      [] as TPricingDto[],
```

Comment:
Can we use satisfies ? I don't know if we are up to date to use it

Reply:
Unfortunately no, our `prettier` version does not allow this usage